src/akushon/action/model/action.cpp

```diff
@@ -108,6 +107,30 @@ const std::string & Action::get_next_action() const
 void Action::reset()
 {
   poses.clear();
+  joint_splines.clear();
 }
 
+void Action::enable_spline(bool enable)
+{
+  use_spline = enable;
+}
+
+bool Action::is_using_spline() const
+{
+  return use_spline;
+}
+
+void Action::generate_splines()
+{
+  joint_splines.clear();
+  for (auto pose : poses) {
+    for (auto & joint : pose.get_joints()) {
+      uint8_t joint_id = joint.get_id();
+      if (joint_splines.find(joint_id) == joint_splines.end()) {
+        joint_splines[joint_id] = std::make_shared<keisan::SmoothSpline>().get();
```

Comment:
It seems that this line will ended up causing null pointer reference error. Store the `joint_splines` with `std::shared_ptr<keisan::SmoothSpline>` instead of with `keisan::SmoothSpline*`:
```suggestion
        joint_splines[joint_id] = std::make_shared<keisan::SmoothSpline>();
```